pkg/client/leaderelectionconfig/config.go

```diff
@@ -33,7 +33,7 @@ const (
 
 func DefaultLeaderElectionConfiguration() componentconfig.LeaderElectionConfiguration {
```

Comment:
Shouldn't the defaults be produced by running the defaulter on a raw object? This function can easily skew from what the defaulter produces, and the defaulter should be the source of truth.

Additionally, there should not be a concept of an internal object having defaults. Defaults are part of versioned APIs.

I'd expect something like:
```
versioned := &componentconfigv1alpha1.LeaderElectionConfiguration{}
scheme.Default(versioned)
internal := &componentconfig.LeaderElectionConfiguration{}
if err := scheme.Convert(versioned, internal, nil); err != nil {
	return nil, err
}
return internal, nil
```

Reply:
@mtaufen Done!

Reply:
You don't have to do it in this PR, but I noticed the pattern of "defaults" divorced from the defaulters is also shared by `k8s.io/kubernetes/cmd/controller-manager/app/options.NewDefaultControllerManagerComponentConfig`, so it would be a good thing to clean that one up too.

Reply:
Ack @mtaufen i got the hang of it, will give it a shot in another PR

Reply:
Hmm, actually, now that I look more closely, I guess `LeaderElectionConfiguration` is just a substructure of the `KubeControllerManagerConfiguration`?

It is a bit strange to have a defaulter for a substructure like this. Without the context of *which* component it belongs to, it's premature to set a default. Do we need to merge this PR asap to fix failing tests or can we clean up the defaulting for `KubeControllerManagerConfiguration` here too?

Reply:
@mtaufen Done! i've cleaned up the defaulting of `KubeControllerManagerConfiguration`